src/mame/olympia/portrait.cpp

```diff
@@ -117,14 +123,16 @@ void portrait_state::ctrl_w(uint8_t data)
 	output().set_value("photo", (data >> 7) & 1);
 }
 
+// $9235-$9236 raw scroll values up to 512
+// $9236 bit 0 defines which of these are used
 void portrait_state::positive_scroll_w(uint8_t data)
 {
 	m_scroll = data;
 }
 
 void portrait_state::negative_scroll_w(uint8_t data)
 {
-	m_scroll = - (data ^ 0xff);
+	m_scroll = 511 - (data ^ 0xff);
```

Comment:
This is effectively now `m_scroll = 511 - (255 - data);`  – why not just express it as `m_scroll = 256 + data;` to make it clearer?

Reply:
Sure, may even be joined with the other port, so that 
```
void portrait_state::scroll_w(offs_t offset, uint8_t data)
{
  m_scroll = data;
  if (offset & 1)
       m_scroll += 256;
}
```